charset_reader.go

```diff
@@ -0,0 +1,38 @@
+package excelize
+
+import (
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+
+	"golang.org/x/text/encoding"
+	"golang.org/x/text/encoding/charmap"
+)
+
+// CharsetReader Decoder from all codepages to UTF-8
+func CharsetReader(charset string, input io.Reader) (rdr io.Reader, err error) {
```

Comment:
Can we use [charset.NewReaderLabel](https://godoc.org/golang.org/x/net/html/charset#NewReaderLabel) instead of the `CharsetReader` function?

Just like this:

```go
decoder.CharsetReader = func(c string, i io.Reader) (io.Reader, error) {
    return charset.NewReaderLabel(c, i)
}
```

Reply:
I did not check the correctness of the search for encodings used in NewReaderLabel without name conversion. With a probability of 99%, no encoding will be found and thus the functionality will be useless. Moreover, some of the names of the encodings used in the charset library do not correspond to the canonical names in the RFC/ISO. I even thought to make synonyms for the correct search for the encoding by name.
Your proposed option
1. Creates a lot of redundant code due to the need to use an anonymous function - a wrapper
2. Will stop working for 90% of encodings

Reply:
If you do not like the presence of CharsetReader inside the library, then this can be fixed by making it possible to injects CharsetReader from the outside.

Adding type:
```
type CharsetTranscoderFn func(charset string, input io.Reader) (rdr io.Reader, err error)
```

Adding function:
```
func OpenReader(fn CharsetTranscoderFn) *File
```

And use this:
```
obj, err := excelize.CharsetTranscoder(fn CharsetTranscoderFn).OpenFile(filename)
```